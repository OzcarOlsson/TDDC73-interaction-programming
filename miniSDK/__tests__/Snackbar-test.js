import React from "react";
import renderer from "react-test-renderer";
import Snackbar from "../components/Snackbar";

// JEST testing
jest.useFakeTimers();

it("renders correctly", () => {
	const tree = renderer.create(<Snackbar />).toJSON();
	expect(tree).toMatchSnapshot();
});

// Warning Snackbar
it("Style prop = warning generates a yellow warning snackbar with 'This is warning message' ", () => {
	const tree = renderer.create(<Snackbar style={"warning"} />).toJSON();
	expect(tree.props.style.backgroundColor).toBe("#e69010");
	expect(tree.children[0].children[0].children[0]).toBe("This is warning message");
});

// Default complete snackbar
it("No style prop provided generates a green warning snackbar with 'This is complete message' ", () => {
	const tree = renderer.create(<Snackbar />).toJSON();
	expect(tree.props.style.backgroundColor).toBe("green");
	expect(tree.children[0].children[0].children[0]).toBe("This is complete message");
});

//Test error snackbar
it("Type warning generates a red warning snackbar with 'This is error message' ", () => {
	const tree = renderer.create(<Snackbar style={"error"} />).toJSON();
	expect(tree.props.style.backgroundColor).toBe("#941212");
	expect(tree.children[0].children[0].children[0]).toBe("This is error message");
});

//Test info snackbar with custom color
it("Type info and a custom color generates a orange snackbar with 'This is info message' ", () => {
	const tree = renderer.create(<Snackbar snackColor={"orange"} style={"info"} />).toJSON();
	expect(tree.props.style.backgroundColor).toBe("orange");
	expect(tree.children[0].children[0].children[0]).toBe("This is info message");
});

// Test if hide button is visible when set to true
it("Snackbar with Hide button set to true", () => {
	const tree = renderer.create(<Snackbar hideButton={true} />).toJSON();

	expect(tree.children[1]).toBeDefined();
});

// Test if hide button is not definied when default = false
it("Snackbar with default hide button is false, should not be defined ", () => {
	const tree = renderer.create(<Snackbar />).toJSON();
	// console.log(tree.children[1]);
	expect(tree.children[1]).toBeUndefined();
});

//Test if own message is displayed correctly
it("Message prop is passed into the component, should display what is stated in the prop ", () => {
	const tree = renderer.create(<Snackbar style={"error"} message={"Fill out all fields"} />).toJSON();
	// console.log(tree.children[0].children[0].children[0]);
	expect(tree.children[0].children[0].children[0]).toBe("Fill out all fields");
});
