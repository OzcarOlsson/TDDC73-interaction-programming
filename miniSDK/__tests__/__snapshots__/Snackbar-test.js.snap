// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "green",
      "borderRadius": 5,
      "bottom": 20,
      "flexDirection": "row",
      "height": 70,
      "justifyContent": "center",
      "padding": 10,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": 500,
        },
      ],
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexWrap": "wrap",
        "maxHeight": "90%",
        "maxWidth": "80%",
        "overflow": "hidden",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "textAlign": "center",
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      This is complete message
    </Text>
  </View>
</View>
`;
